### ISSUE TITLE
Suggestions for applying to visual dense prediction tasks.

### ISSUE THREAD

**Issue by pUmpKin-Co** [sail-sg/Adan#21]
HI~Thanks for you excitring work. I would like to know the performance of  Adan for visual dense prediction tasks. I notice you mention that Adan is suitable for large batchsize. So I wondered if it would also work better for visual dense prediction tasks, which are usually not possible with a large batchsize. I have tried Adan in several tasks, but the results are similar or even inferior to its sgd/adamw counterparts. I have followed the best practices you mention in the paper and repo and was wondering if you have done similar experiments or if you have suggestions for tuning the parameters.

Thanks!

Best.

**Comment by XingyuXie** [comment]
Hi @pUmpKin-Co,
You may send me the details of the settings/tasks; I will check the hyperparameters. If the tasks are the same as our released experiments. The results are reproducible.

If you focus on the small batch size, you could show the hyper-parameter for AdamW, SGD, and Adan here. I will help you to find the proper LR and weight-decay.

**Comment by pUmpKin-Co** [comment]
Thanks! 

Semantic Segmentation:
```
bs: 16

ResNet50:
sgd-lr: 0.01
sgd-wd: 0.0001
result (mIoU):  89.07

adan-lr: 0.0003 (5-10 times of adamw) result: 87.05 or 0.0015 (\sqrt(16/256) * 0.00625) result: NaN
adan-wd: 0.02
adan-grad-norm: 5.0

swin-b:
adamw: 0.00006
adamw-wd: 0.05
result: 89.51

adan-lr: 0.0003
adan-wd: 0.02
adan-grad-norm: 0.
result: 88.10
```

Detection
```
bs: 8

ResNet50:
sgd-lr: 0.02
sgd-wd: 0.0001
result (mAP): 76.21

adan-lr: 0.0005 (5-10 times of adamw) result: 76.01 or 0.0011 (\sqrt(16/256) * 0.00625) result: 74.84
adan-wd: 0.02
adan-grad-norm: 5.0

swin-b:
adamw: 0.0001
adam-wd: 0.05
result: 76.65

adan-lr: 0.0005
adan-wd: 0.02
adan-grad-norm: 0.
result: 74.21
```
I also found that when using adan, the accuracy improved faster than other optimizers in the early stages of training, but the results were not as good. 

The learning rate scheduler is CosineAnnealing.

**Comment by XingyuXie** [comment]
Hi @pUmpKin-Co,
Did you use the warmup LR strategy in your settings? For larger LR, we should utilize larger warmup steps.
Your setting seems okay, and I suggest focusing on the swin-b cases. If there are no warmup steps, you may decay the LR; for example, set `adan-lr = 0.00015`. For this stage of finding proper LR, I suggest fixing `wd = 0.02`

**Comment by XingyuXie** [comment]
@pUmpKin-Co 
I will try to finetune Detectron-2 with Adan and will report the results and the hyper-parameter here.

**Comment by pUmpKin-Co** [comment]
Hi! Following your suggestions, I have tried longer warmup step (x2). The results are comparable with other conterparts.  I think perhaps for visually dense prediciton tasks, a lower learning rate or longer warmup may be required.

**Comment by XingyuXie** [comment]
@pUmpKin-Co 
You may further increase the warmup step and find the proper one for the x5 times LR. By the way, for the classification problem, the `min_lr` is an important parameter, and it depends on the data and model size. You may try to tune it.

I will also report some results/logs/settings on ADE20K in the next few days.
